src/webui/webui.cpp

```diff
@@ -145,7 +146,41 @@ void WebUI::configure()
     }
 }
 
+bool WebUI::isEnabled() const
+{
+    return m_isEnabled;
+}
+
 bool WebUI::isErrored() const
 {
     return m_isErrored;
 }
+
+QString WebUI::scheme() const
+{
+    if (!m_httpServer) return {};
+    return m_httpServer->isHttps() ? u"https"_s : u"http"_s;
+}
+
+QString WebUI::host() const
+{
+    if (!m_httpServer) return {};
+    const auto address = m_httpServer->serverAddress();
+    return (address == QHostAddress::Any) ? u"localhost"_s : address.toString();
```

Comment:
>Look now, should be much cleaner.

You probably still didn't understand the essence of my claims, since this logic is still here. It is incorrect as property of WebUI. It is generally incorrect in the general case, since `QHostAddress::Any` is not equal to `localhost`. The appropriate `WebUI` property should return exactly the used/assigned host address.
If it really makes sense to substitute values like `QHostAddress::Any` (note that there are also other similar values exist) by "localhost" in some particular case, then this should be done at the place of use.

Reply:
Hmm, yes, I didn't realized that. Fair enough, I will rework it.

Reply:
How about now?